apps/src/applab/ImportScreensDialog.jsx

```diff
@@ -0,0 +1,407 @@
+    var {screen} = this.props;
+    return (
+      <div style={[styles.screenListItem,
+                   !screen.canBeImported && styles.disabledScreenListItem]}>
+        <div style={styles.miniScreenWrapper}>
+          <div dangerouslySetInnerHTML={{__html: screen.html}}
+               style={styles.miniScreen}
+          />
+        </div>
+        <div>
+          {screen.id}
+          {screen.willReplace &&
+           <p style={styles.warning}>
+             Importing this will replace your existing screen: "{screen.id}".
+           </p>}
+           {screen.assetsToReplace.length > 0 &&
+            <p style={styles.warning}>
+              Importing this will replace your existing
+              assets: {screen.assetsToReplace.map(a => `"${a}"`).join(', ')}.
+            </p>
+           }
+            {screen.conflictingIds.length > 0 &&
+             <p style={styles.warning}>
+               Uses existing element IDs: {screen.conflictingIds.map(id => `"${id}"`).join(', ')}.
+             </p>}
+        </div>
+      </div>
+    );
+  }
+}));
+
+var ImportScreensDialog = React.createClass({
+
+  propTypes: Object.assign({}, Dialog.propTypes, {
+    project: importableProjectShape.isRequired,
+    onImport: React.PropTypes.func.isRequired,
+  }),
+
+  getInitialState() {
+    return {
+      selectedScreens: [],
+      selectedAssets: [],
+    };
+  },
+
+  render() {
+    const nonImportableScreens = this.props.project.screens.filter(s => !s.canBeImported);
+    return (
+      <Dialog
+          title={`Import from Project: ${this.props.project.name}`}
+          confirmText="Import"
+          onConfirm={() => this.props.onImport(this.state.selectedScreens,
+                                               this.state.selectedAssets)}
+          {...this.props}>
+        <Body>
+          <MultiCheckboxSelector
+              style={styles.section}
+              header="Screens"
+              items={this.props.project.screens.filter(s => s.canBeImported)}
+              selected={this.state.selectedScreens}
+              onChange={selectedScreens => this.setState({selectedScreens})}
+              itemPropName="screen">
+            <ScreenListItem />
+          </MultiCheckboxSelector>
+          {this.props.project.otherAssets.length > 0 &&
+           <MultiCheckboxSelector
+               style={styles.section}
+               header="Other Assets"
+               items={this.props.project.otherAssets}
+               selected={this.state.selectedAssets}
+               onChange={selectedAssets => this.setState({selectedAssets})}
+               itemPropName="asset">
+             <AssetListItem/>
+           </MultiCheckboxSelector>}
+          {nonImportableScreens.length > 0 &&
+           <div style={styles.section}>
+             <h2 style={multiCheckboxStyles.header}>Cannot Import</h2>
+             <p style={styles.subtext}>
+               Cannot import the following screens because they contain design elements
```

Comment:
Applab does get localized, and this will need to change. I'd like to take a closer look out how we do i18n in general before deciding how to hook it up here. So this is a todo.